tfx/examples/iris/iris_utils.py

```diff
@@ -206,23 +208,46 @@ def run_fn(fn_args: executor.TrainerFnArgs):
   """
   schema = io_utils.parse_pbtxt_file(fn_args.schema_file, schema_pb2.Schema())
 
-  training_spec = trainer_fn(fn_args, schema)
+  # TODO(b/160795287): Deprecate estimator based executor.
+  # Modify fn_args with model_run_dir given as the working directory.
+  # Executor will then copy user models to model artifact directory.
+  fn_args_dict = fn_args.get_dict()
+  fn_args_dict['serving_model_dir'] = os.path.join(fn_args.model_run_dir,
+                                                   path_utils.SERVING_MODEL_DIR)
+  fn_args_dict['eval_model_dir'] = os.path.join(fn_args.model_run_dir,
+                                                path_utils.EVAL_MODEL_DIR)
+  user_fn_args = executor.TrainerFnArgs(arg_dict=fn_args_dict)
+  training_spec = trainer_fn(user_fn_args, schema)
```

Comment:
trainer_fn is not used by tfx (it should be _trainer_fn) in this example, you can direct update trainer_fn